pkg/scheduler/service/service.go

```diff
@@ -473,14 +473,25 @@ func (s *SchedulerServer) awaitFailover(schedulerProcess schedulerProcessInterfa
 
 func validateLeadershipTransition(desired, current string) {
 	log.Infof("validating leadership transition")
-	d := uid.Parse(desired).Group()
-	c := uid.Parse(current).Group()
-	if d == 0 {
+	desiredUID, err := uid.Parse(desired)
+	if err != nil {
+		log.Fatalf("Failed to parse desired scheduler UID %q: %v", desired, err)
+	}
+
+	desiredGroup := desiredUID.Group()
+	if desiredGroup == 0 {
 		// should *never* happen, but..
 		log.Fatalf("illegal scheduler UID: %q", desired)
 	}
-	if d != c && c != 0 {
-		log.Fatalf("desired scheduler group (%x) != current scheduler group (%x)", d, c)
+
+	currentUID, err := uid.Parse(desired)
```

Comment:
you've changed the semantics here. the 0 current-group (c) was allowed before. Parse returns an error instead of nil now and the code fatally exits on that condition. so this func will behave differently. desired action via truth table:
```
d | c | result
---------------
0 | 0 | fail
0 | 1 | fail
0 | 2 | fail
1 | 0 | c was nil (no current leader); OK
1 | 1 | d == c; OK
1 | 2 | d != c && c != 0; fail
2 | 0 | c was nil (no current leader); OK
2 | 1 | d != c && c != 0; fail
2 | 2 | d == c; OK
```